storm-shim/src/main/java/storm/mesos/shims/ICommandLineShim.java

```diff
@@ -0,0 +1,12 @@
+package storm.mesos.shims;
+
+/**
+ * Returns the command line, shimmed and 0.9.X and 0.10.X have slightly a different command
+ */
+public interface ICommandLineShim {
+
+  public String getCommandLine();
+
+  public String getDockerCommandLine();
```

Comment:
I think going with the factory approach is just a start for refactoring `computeResourcesForSlot()`.  That method does way more than just compute the resources.

For the near term, the `MesosNimbus` code could go from:
```java
        if (_container.isPresent()) {
          // An ugly workaround for a bug in DCOS
          Map<String, String> env = System.getenv();
          String javaLibPath = env.get("MESOS_NATIVE_JAVA_LIBRARY");

          executorInfoBuilder
              .setCommand(CommandInfo.newBuilder()
                  .addUris(URI.newBuilder().setValue(configUri))
                  .setValue(
                      "export MESOS_NATIVE_JAVA_LIBRARY=" + javaLibPath +
                          " && /bin/cp $MESOS_SANDBOX/storm.yaml conf && /usr/bin/python bin/storm " +
                          "supervisor storm.mesos.MesosSupervisor -c storm.log.dir=$MESOS_SANDBOX/logs" + extraConfig))
              .setContainer(
                  ContainerInfo.newBuilder()
                      .setType(ContainerInfo.Type.DOCKER)
                      .setDocker(
                          ContainerInfo.DockerInfo.newBuilder()
                              .setImage(_container.get())
                              .setNetwork(ContainerInfo.DockerInfo.Network.HOST)
                              .setForcePullImage(true)
                              .build()
                      )
                      .build()
            );
        } else {
          executorInfoBuilder
              .setCommand(CommandInfo.newBuilder()
                  .addUris(URI.newBuilder().setValue((String) _conf.get(CONF_EXECUTOR_URI)))
                  .addUris(URI.newBuilder().setValue(configUri))
                  .setValue("cp storm.yaml storm-mesos*/conf && cd storm-mesos* && python bin/storm " +
                      "supervisor storm.mesos.MesosSupervisor" + extraConfig));
        }
```
to
```java
    ICommandLineShim  commandLineShim = Factory.makeCommandLineShim(commandType);
          executorInfoBuilder
              .setCommand(CommandInfo.newBuilder()
                  .addUris(URI.newBuilder().setValue((String) _conf.get(CONF_EXECUTOR_URI)))
                  .addUris(URI.newBuilder().setValue(configUri))
                  .setValue(commandLineShim.getCommandLine()));
        if (_container.isPresent()) {
          executorInfoBuilder
              .setContainer(
                  ContainerInfo.newBuilder()
                      .setType(ContainerInfo.Type.DOCKER)
                      .setDocker(
                          ContainerInfo.DockerInfo.newBuilder()
                              .setImage(_container.get())
                              .setNetwork(ContainerInfo.DockerInfo.Network.HOST)
                              .setForcePullImage(true)
                              .build()
                      )
                      .build()
            );
        }
```

I don't have a strong opinion, but I do think that future refactoring will make it so the more OO design will be used.